src/libaktualizr/uptane/uptane_network_test.cc

```diff
@@ -139,6 +140,37 @@ TEST(UptaneNetwork, DownloadFailure) {
   EXPECT_TRUE(result.first);
 }
 
+/*
+ * Output a log when connectivity is restored.
+ */
+TEST(UptaneNetwork, LogConnectivityRestored) {
+  testing::internal::CaptureStdout();
+  TemporaryDirectory temp_dir;
+  conf.storage.type = StorageType::kSqlite;
+  conf.storage.path = temp_dir.Path();
+  conf.provision.expiry_days = "noerrors";
+  conf.provision.primary_ecu_serial = "noerrors";
+  const std::string good_url = conf.provision.server;
+
+  auto http = std::make_shared<HttpClient>();
+  auto store = INvStorage::newStorage(conf.storage);
+
+  auto sota_client = std_::make_unique<SotaUptaneClient>(conf, store, http);
+  EXPECT_NO_THROW(sota_client->initialize());
+  EXPECT_NO_THROW(sota_client->fetchMeta());
+  EXPECT_NO_THROW(sota_client->checkUpdates());
```

Comment:
If you're using the SotaUptaneClient functions directly, you probably want to just call `initialize` once and then `fetchMeta` each time. `fetchMeta` is actually what `Aktualizr::CheckUpdates` calls, and `fetchMeta` calls `SotaUptaneClient::checkUpdates`. It also is what calls `putManifestSimple`, which is where your change is located, so just calling `SotaUptaneClient::checkUpdates` won't trigger it.

Reply:
BTW if you make this test case a `FRIEND_TEST` of `SotaUptaneClient` (see `sotauptaneclient.h` for other examples) you can directly manipulate the `Config` object stored in `SotaUptaneClient` and then you can bypass needing to re-instantiate or re-initialize `SotaUptaneClient`.